django/core/paginator.py

```diff
@@ -103,8 +103,23 @@ def _check_object_list_is_ordered(self):
                 UnorderedObjectListWarning
             )
 
+    def effective_page(self, number):
```

Comment:
Thoughts about a name like `get_page` or something with `safe` in it? I don't think "effective" has much meaning here, it's really about the error handling.

I see that there is already a `_get_page` function (although the fact that it is a hook according to its docstring means it should probably be public). What about putting the logic in there?